### PR TITLE
Add provider to host the env vars shared by all containers

### PR DESCRIPTION
**Pull request by fvaleri**
The environment variables shared by all containers are statically defined in ContainerUtils. They include proxy env vars that are also set in the Dockerfile for KafkaConnect builds.

When proxy env vars are set or updated, some tests fail due to the following reasons:

The Dockerfile hash changes:

- KafkaConnectAssemblyOperatorPodSetTest
- KafkaConnectBuildAssemblyOperatorKubeTest
- KafkaConnectBuildAssemblyOperatorOpenShiftTest

The expected test result changes:

- KafkaConnectClusterTest
- KafkaExporterTest
- EntityTopicOperatorTest
- EntityUserOperatorTest
- KafkaBridgeClusterTest
- KafkaMirrorMaker2ClusterTest
- KafkaMirrorMakerClusterTest
- CruiseControlTest

This change adds the SharedEnvironmentProvider interface with a DefaultSharedEnvironmentProvider implementation. This is propagated to all components, and can be replaced with a MockSharedEnvironmentProvider for testing. The mock implementation makes the unit and integration test suites independent from the test environment. Additionally, this completely replace the previous fix which only addresses the second issue.

Unit and integration tests result on my machine: 

```
[INFO] Reactor Summary for Strimzi - Apache Kafka on Kubernetes and OpenShift 0.36.0-SNAPSHOT:
[INFO] 
[INFO] Strimzi - Apache Kafka on Kubernetes and OpenShift . SUCCESS [  1.572 s]
[INFO] kafka-agent ........................................ SUCCESS [  1.317 s]
[INFO] mirror-maker-agent ................................. SUCCESS [  0.063 s]
[INFO] tracing-agent ...................................... SUCCESS [  0.088 s]
[INFO] test ............................................... SUCCESS [  0.260 s]
[INFO] crd-annotations .................................... SUCCESS [  0.526 s]
[INFO] crd-generator ...................................... SUCCESS [  3.295 s]
[INFO] api ................................................ SUCCESS [01:56 min]
[INFO] mockkube ........................................... SUCCESS [  3.209 s]
[INFO] config-model ....................................... SUCCESS [  0.483 s]
[INFO] config-model-generator ............................. SUCCESS [ 14.324 s]
[INFO] certificate-manager ................................ SUCCESS [ 16.735 s]
[INFO] operator-common .................................... SUCCESS [03:43 min]
[INFO] topic-operator ..................................... SUCCESS [09:44 min]
[INFO] cluster-operator ................................... SUCCESS [15:04 min]
[INFO] user-operator ...................................... SUCCESS [ 37.854 s]
[INFO] kafka-init ......................................... SUCCESS [  2.737 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  31:51 min
[INFO] Finished at: 2023-06-01T14:53:23+02:00
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by scholzj** [comment]
@fvaleri Can you please rebase this when you have time?

**Comment by fvaleri** [comment]
The branch is rebased and I also re-ran all unit and integrations tests with and without NO_PROXY exported (the trigger of the original issue):

```sh
[INFO] Reactor Summary for Strimzi - Apache Kafka on Kubernetes and OpenShift 0.36.0-SNAPSHOT:
[INFO] 
[INFO] Strimzi - Apache Kafka on Kubernetes and OpenShift . SUCCESS [  2.807 s]
[INFO] kafka-agent ........................................ SUCCESS [  3.872 s]
[INFO] mirror-maker-agent ................................. SUCCESS [  1.264 s]
[INFO] tracing-agent ...................................... SUCCESS [  1.196 s]
[INFO] test ............................................... SUCCESS [  2.350 s]
[INFO] crd-annotations .................................... SUCCESS [  1.527 s]
[INFO] crd-generator ...................................... SUCCESS [  4.608 s]
[INFO] api ................................................ SUCCESS [02:15 min]
[INFO] mockkube ........................................... SUCCESS [  4.415 s]
[INFO] config-model ....................................... SUCCESS [  1.379 s]
[INFO] config-model-generator ............................. SUCCESS [ 13.506 s]
[INFO] certificate-manager ................................ SUCCESS [ 17.151 s]
[INFO] operator-common .................................... SUCCESS [03:36 min]
[INFO] topic-operator ..................................... SUCCESS [09:41 min]
[INFO] cluster-operator ................................... SUCCESS [11:47 min]
[INFO] user-operator ...................................... SUCCESS [ 38.944 s]
[INFO] kafka-init ......................................... SUCCESS [  3.719 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  28:57 min
[INFO] Finished at: 2023-06-15T13:25:42+02:00
[INFO] ------------------------------------------------------------------------
```

File: operator-common/src/main/java/io/strimzi/operator/common/SharedEnvironmentProvider.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        FIPS_MODE;
+
+        /**
+         * Returns the list of env var names.
+         * @return list of env var names
+         */
+        public static List<String> names() {
+            return Arrays.stream(EnvVarName.values()).map(Enum::name).toList();
+        }
+    }
```
Comment (scholzj) [inline operator-common/src/main/java/io/strimzi/operator/common/SharedEnvironmentProvider.java:38]: I wonder if this should be really in the interface? Why not make the method abstract and move it into the `DefaultSharedEnvironmentProvider` class?

File: operator-common/src/main/java/io/strimzi/operator/common/SharedEnvironmentProvider.java
Comment (scholzj) [inline operator-common/src/main/java/io/strimzi/operator/common/SharedEnvironmentProvider.java:1]: You do not use it anywhere outside the CO. And there do not seem to be any use case outside of CO. So I think you should move it to the CO module as well.

**Review by scholzj — COMMENTED** [review]
Look mostly good to me. I left a comment about placement of the class and one other minor thing.

File: operator-common/src/main/java/io/strimzi/operator/common/SharedEnvironmentProvider.java (hunk 1 above)
Comment (fvaleri) [reply operator-common/src/main/java/io/strimzi/operator/common/SharedEnvironmentProvider.java:38]: Yes, I think it's better. Thanks.

**Comment by fvaleri** [comment]
@scholzj thanks for the review, I added the changes you suggested. I think the failing regression test was unrelated.

@ppatierno @tombentley I touched a lot of classes, but the proposed change is simple. I'm just adding a new provider for environment variables that are shared by most components.

**Review by scholzj — APPROVED** [review]
LGTM. I will rerun the STs as one of them failed last time. But I think it was unrelated.

**Review by ppatierno — APPROVED** [review]
LGTM

**Comment by scholzj** [comment]
@tombentley Can you please have a look? I think it would be good to have your OK for this PR.

**Review by tombentley — APPROVED** [review]

**Pull request merged**